frontend/src/store-logic.ts

```diff
@@ -24,11 +24,16 @@ function isUpdateResponse(p: any): p is UpdateResponse {
   return p.type === 'Update'
 }
 
-export function to_app_infos_by_id<T extends { app_id: string }>(app_infos: T[]): Record<string, T> {
-  return app_infos.reduce((acc, appinfo) => {
-    acc[appinfo.app_id] = appinfo
-    return acc
-  }, {} as Record<string, T>)
+function isComplete(elements: Record<string, Partial<AppInfo> | null>): elements is Record<string, AppInfo> {
+  for (const element of Object.values(elements)) {
+    if (element == null) {
```

Comment:
Yeah was a bit sloppy with that case. It will never happen so I just added that for typechecking reasons. A false return value would be better here indeed. Also, === is better there